### PR TITLE
Test with clojure versions 1.8, 1.9, 1.10.*

### PR DESCRIPTION
**Pull request by schmir**
Add profiles to project.clj for those clojure versions. Add test-all
and deps-all aliases in order to make it easy to test locally.

Also test with each of these versions on circleci. Use a dedicated
test step for each version.

builds on circleci can be viewed here:

https://circleci.com/gh/schmir/ring-buffer/17

https://circleci.com/gh/schmir/ring-buffer/18

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,25 +8,37 @@ commands:
       # Download and cache dependencies
       - restore_cache:
           keys:
-            - v1-dependencies-{{ checksum "project.clj" }}
+            - v2-dependencies-{{ checksum "project.clj" }}
```
Comment (danielcompton) [inline .circleci/config.yml]: We don't need to bump the version here. You only need to bump to v2 if you drastically change things in a way where the `project.clj` itself doesn't change.

**Review by danielcompton — CHANGES_REQUESTED** [review]
Thanks, one nit.

File: .circleci/config.yml (hunk 1 above)
Comment (schmir) [reply .circleci/config.yml]: I've changed the command that fetches the dependencies outside of the project.clj file. But you're right, I've also changed the project.clj file.

Comment (schmir) [reply .circleci/config.yml]: I've force pushed a new version that doesn't change the key.

**Comment by danielcompton** [comment]
Thanks!

**Pull request merged**